probe-rs/src/flash/error.rs

```diff
@@ -0,0 +1,50 @@
+use thiserror::Error;
+
+use crate::config::FlashRegion;
+use crate::error;
+
+#[derive(Error, Debug)]
+pub enum FlashError {
+    #[error("The execution of '{name}' failed with code {errorcode}")]
+    RoutineCallFailed { name: &'static str, errorcode: u32 },
+    #[error("'{0}' is not supported")]
+    NotSupported(&'static str),
```

Comment:
What exactly is not supported? This seems a bit generic to me, what does the `str` describe?

Reply:
I have no clue actually :D I think it is meant so the developer can return whatever they want.

Reply:
Fixed.